tests/unit/SpecialResolution.spec.ts

```diff
@@ -0,0 +1,367 @@
+
+    // Validate Office Addresses
+    expect(state.officeAddresses.registeredOffice.deliveryAddress.streetAddress)
+      .toBe('reg delivery_address - address line one')
+    expect(state.officeAddresses.recordsOffice.mailingAddress.streetAddress)
+      .toBe('rec mailing_address - address line two')
+
+    // Validate People And Roles
+    expect(store.state.stateModel.peopleAndRoles.orgPeople[0].officer.firstName).toBe('USER')
+    expect(store.state.stateModel.peopleAndRoles.orgPeople[0].officer.lastName).toBe('ONE')
+    expect(store.state.stateModel.peopleAndRoles.orgPeople[0].roles[0].roleType).toBe('Director')
+
+    // Validate Contact Info
+    expect(store.state.stateModel.businessContact.email).toBe('mock@example.com')
+    expect(store.state.stateModel.businessContact.phone).toBe('123-456-7890')
+
+    expect(store.state.stateModel.currentFees.filingFees).toBe(70)
+    expect(store.state.stateModel.currentFees.futureEffectiveFees).toBe(0)
+  })
+
+  it('fetches the fee prices after loading', async () => {
+    await wrapper.setProps({ appReady: true })
+    await flushPromises()
+    expect(store.state.stateModel.feePrices.filingFees).toBe(70)
+    expect(store.state.stateModel.feePrices.futureEffectiveFees).toBe(0)
+  })
+
+  it('updates the current fees when SpecialResolutionSummary changes', async () => {
+    await wrapper.setProps({ appReady: true })
+    await flushPromises()
+
+    const state = store.state.stateModel
+    state.effectiveDateTime.isFutureEffective = true
+
+    await wrapper.vm.onSpecialResolutionSummaryChanges()
+    expect(store.state.stateModel.currentFees.filingFees).toBe(70)
+    expect(store.state.stateModel.currentFees.futureEffectiveFees).toBe(0)
+  })
+
+  it('certify text is not prefilled/editable for staff user', async () => {
+    store.state.stateModel.tombstone.keycloakRoles = ['staff']
+    store.state.stateModel.tombstone.userInfo = {
+      firstname: 'Jon',
+      lastname: 'Doe'
+    }
+    await wrapper.setProps({ appReady: true })
+    await flushPromises()
+
+    expect(store.state.stateModel.certifyState.certifiedBy).toBe('undefined undefined')
+  })
+
+  it('certify text is prefilled/uneditable for non-staff user', async () => {
+    store.state.stateModel.tombstone.keycloakRoles = []
+    store.state.stateModel.tombstone.userInfo = {
+      firstname: 'Jon',
+      lastname: 'Doe'
+    }
+    await wrapper.setProps({ appReady: true })
+    await flushPromises()
+
+    expect(store.state.stateModel.certifyState.certifiedBy).toBe('Jon Doe')
+  })
+
+  // FUTURE
+  xit('loads a draft alteration into the store', async () => {
+    // Validate Effective Date-Time
+    expect(store.state.stateModel.effectiveDateTime.isFutureEffective).toBe(true)
+    expect(store.state.stateModel.effectiveDateTime.dateTimeString).toBe('2021-03-22T18:00:00.000Z')
+    expect(store.state.stateModel.validationFlags.flagsReviewCertify.isValidEffectiveDate).toBe(true)
+  })
```

Comment:
I know that we've left some other unit tests incomplete, but if you have time then it would be great to finish this unit tests, since it will verify loading data from a draft versus the business snapshot, which is important functionality.

Reply:
I will add more test case when we are working on components